src/entries/popup/hooks/useHomeShortcuts.ts

```diff
@@ -73,6 +76,57 @@ export function useHomeShortcuts() {
     });
   }, [address]);
 
+  interface WalletObj {
+    accounts: Address[];
+    type: 'HdKeychain' | 'HardwareWalletKeychain' | 'ReadOnlyKeychain'; // Add other possible types if there are more
+    vendor?: string;
+  }
+
+  const generateCSV = useCallback(
+    (data: WalletObj[]) => {
+      let csvContent = 'public_address,name,type\n';
+
+      data.forEach(
+        (item: { accounts: string[]; type: string; vendor?: string }) => {
+          const address = item.accounts[0];
+          // If address exists in walletNames, use that name. Otherwise, use defaultName.
+          const name = walletNames[address as Address] || item.accounts[0];
+          const importedType =
+            item.type === 'ReadOnlyKeychain' ? 'Watching' : 'Imported';
+          const type = item.vendor || importedType;
+
+          csvContent += `${address},${name},${type}\n`;
+        },
+      );
+
+      return csvContent;
+    },
+    [walletNames],
+  );
+
+  const handleCopyList = useCallback(async () => {
+    console.log('walletNames', walletNames);
+    const data = await getWallets();
+    console.log('getWallets', data);
+    const csvContent = generateCSV(data as WalletObj[]);
+    const blob = new Blob([csvContent], { type: 'text/csv' });
+    const url = URL.createObjectURL(blob);
+
+    chrome.downloads.download(
+      {
+        url: url,
+        filename: 'accounts.csv',
+      },
+      () => {
+        URL.revokeObjectURL(url);
+      },
+    );
+    triggerToast({
+      title: 'Address CSV file created',
+      description: 'downloading...',
```

Comment:
"Addresses Downloaded" with the CSV file name as the description would be nice here; just like the Copied toast